app/controllers/posts_controller.rb

```diff
@@ -6,6 +6,30 @@ def index
 
   def show
     @user = User.find(params[:user_id])
-    @post = Post.find_by(id: params[:id], author_id: params[:user_id])
+    @post = Post.find(params[:id])
+    @like = Like.new
+    @comment = Comment.new
+  end
+
+  def new
+    @user = User.find(params[:user_id])
+    @post = Post.new
+  end
+
+  def create
+    @user = User.find(params[:user_id])
+    @post = @user.posts.build(post_params)
+
+    if @post.save
+      redirect_to user_post_path(@user, @post), notice: 'Post created successfully.'
+    else
+      render :new
+    end
+  end
```

Comment:
- [ ] Kindly make sure a new post is created on behalf of the currently logged-in user (current_user) of the app in order to match the project requirements.

```suggestion
  def create
    @user = current_user
    @post = @user.posts.build(post_params)

    if @post.save
      redirect_to user_post_path(@user, @post), notice: 'Post created successfully.'
    else
      render :new
    end
  end
```

Reply:
@levy002 Thanks for your feedback! I have already made the changes and am requesting another review.

Reply:
Hello @PabloBona , thanks for implementing requested changes. Kindly go ahead and submit your PR link in your dashboard, you will receive the feedback in no time. Happy Friday ☺️